datamodel/low/extraction_functions.go

```diff
@@ -106,49 +105,75 @@ func LocateRefNodeWithContext(ctx context.Context, root *yaml.Node, idx *index.S
 			}
 		}
 
-		// perform a search for the reference in the index
-		// extract the correct root
+		// Obtain the absolute filepath/URL of the spec in which we are trying to
```

Comment:
You OK with leaving these doc comments? I put these here when initially stepping through the code. I hope they'd help me or someone else in the future.